nuget/lib/dependabot/nuget/nuget_config_credential_helpers.rb

```diff
@@ -0,0 +1,70 @@
+# typed: true
+# frozen_string_literal: true
+
+module Dependabot
+  module Nuget
+    module NuGetConfigCredentialHelpers
+      def self.user_nuget_config_path
+        home_directory = Dir.home
+        File.join(home_directory, ".nuget", "NuGet", "NuGet.Config")
+      end
+
+      def self.temporary_nuget_config_path
+        user_nuget_config_path + "_ORIGINAL"
+      end
+
+      def self.add_credentials_to_nuget_config(credentials)
+        return unless File.exist?(user_nuget_config_path)
+
+        nuget_credentials = credentials.select { |cred| cred["type"] == "nuget_feed" }
+        return if nuget_credentials.empty?
+
+        File.rename(user_nuget_config_path, temporary_nuget_config_path)
+
+        package_sources = []
+        package_source_credentials = []
+        nuget_credentials.each_with_index do |c, i|
+          source_name = "credentialed_source_#{i + 1}"
+          package_sources << "    <add key=\"#{source_name}\" value=\"#{c.fetch('url')}\" />"
+          package_source_credentials << "    <#{source_name}>"
+          package_source_credentials << "      <add key=\"Username\" value=\"user\" />"
+          package_source_credentials << "      <add key=\"ClearTextPassword\" value=\"#{c.fetch('token')}\" />"
+          package_source_credentials << "    </#{source_name}>"
+        end
+
+        nuget_config = <<~NUGET_XML
+          <?xml version="1.0" encoding="utf-8"?>
```

Comment:
Is it possible that the original nuget.config file had different sections than packagesSources and packageSourceCredentials that are needed for the repo to work?

Reply:
No, this file has a set content that's put down during the install and these are the only things we should need.  We're editing this file because the repo itself (cloned to a different directory) might have its own nuget.config files, but the behavior of nuget is to also patch in the contents of this file which we completely controll.